### PR TITLE
Make sure algolia indexes job posting if action text changes

### LINKED ISSUE #341: JobPosting fix indexing
Due to change to ActionText for the 3 main contents the autoindexing for algolia does not work.

### LINKED ISSUE #347: JobPosting rich text indexing
Due to change to ActionText for the 3 main contents their index changed to a full object

### PR DESCRIPTION
**Pull request by TheFehr**
Closes #341 
Closes #347

### CODE REVIEW

**Comment by lukasbischof** [comment]
Hm, I'm still not 100% sure about this. Seems hacky.

Does it maybe work if you do something like:

```ruby
class JobPosting < ApplicationRecord
  has_rich_text :description
  has_rich_text :required_skills
  has_rich_text :preferred_skills

  after_save_commit :index!, if: -> { description.changed? || required_skills.changed? || preferred_skills.changed? }
end
```

**Comment by TheFehr** [comment]
Well I did it how the official docs tell you too. The only hacky part is how U added the touch to the ActionText

**Comment by lukasbischof** [comment]
Yeah, well that's exactly the hacky part that I mean 😄

File: app/models/concerns/job_posting_searchable.rb [hunk 1]
```diff
@@ -20,9 +19,11 @@ module JobPostingSearchable
...
       end
 
-      attribute :relevancy do
-        relevancy_for_database
-      end
+      attribute(:relevancy) { relevancy_for_database }
+
+      attribute(:description) { description.body }
+      attribute(:required_skills) { required_skills.body }
+      attribute(:preferred_skills) { preferred_skills.body }
```
Comment (lukasbischof) [inline app/models/concerns/job_posting_searchable.rb:26]: does body return with HTML tags?

Comment (TheFehr) [reply app/models/concerns/job_posting_searchable.rb:26]: yep, checked it in local algolia index

File: config/initializers/action_text_job_posting.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    def record_change
+      return unless body_changed?
+      return unless ::ActionTextJobPosting::RICH_TEXT_ATTRIBUTES.include?(name) && record_type == 'JobPosting'
+
+      # rubocop:disable Rails/SkipsModelValidations
+      record.touch
+      # rubocop:enable Rails/SkipsModelValidations
+    end
+  end
+end
```
Comment (lukasbischof) [inline config/initializers/action_text_job_posting.rb:22]: Is not needed anymore?

File: app/models/concerns/job_posting_searchable.rb (hunk 1 above)
Comment (TheFehr) [reply app/models/concerns/job_posting_searchable.rb:26]: or do u not want the html?

File: app/models/concerns/job_posting_searchable.rb [hunk 3]
```diff
@@ -35,6 +36,9 @@ module JobPostingSearchable
       attributeForDistinct 'organization_display_name'
     end
     # :nocov:
+
+    after_save_commit :index!, if:
+      -> { !changed? && (description.changed? || required_skills.changed? || preferred_skills.changed?) }
```
Comment (lukasbischof) [inline app/models/concerns/job_posting_searchable.rb:41]: I'd prefer

```suggestion
    after_save_commit :index!, if: -> { 
      !changed? && (description.changed? || required_skills.changed? || preferred_skills.changed?) 
    }
```

File: app/models/concerns/job_posting_searchable.rb (hunk 1 above)
Comment (lukasbischof) [reply app/models/concerns/job_posting_searchable.rb:26]: No. I don't expect to search for <ul> and actually get results

Comment (TheFehr) [reply app/models/concerns/job_posting_searchable.rb:26]: oh well, me fixed

File: app/models/concerns/job_posting_searchable.rb [hunk 4]
```diff
@@ -20,9 +19,11 @@ module JobPostingSearchable
         Rails.application.routes.url_helpers.job_posting_url(self, host: ENV['APP_HOST'], port: ENV['APP_PORT'])
       end
 
-      attribute :relevancy do
-        relevancy_for_database
-      end
+      attribute(:relevancy) { relevancy_for_database }
+
+      attribute(:description) { description.body&.to_plain_text }
```
Comment (lukasbischof) [inline app/models/concerns/job_posting_searchable.rb:24]: Can `description` maybe also return nil? So that calling `.body` would not work?

**Review by lukasbischof — APPROVED** [review]

File: app/models/concerns/job_posting_searchable.rb (hunk 4 above)
Comment (TheFehr) [reply app/models/concerns/job_posting_searchable.rb:24]: Good question

Comment (TheFehr) [reply app/models/concerns/job_posting_searchable.rb:24]: But I believe no. I checked a job posting with a known empty rich text and algolia index shows a null

**Pull request merged**